### DISCUSSION TITLE
🚀 Feature Request: Color palette themes and swapping them. Light/Dark Mode and more.

### DISCUSSION THREAD

**Discussion by 23Chromosomes** [elementor/elementor#21161] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Dark mode/Light mode. + Optional color blind mode and custom.
Currently, there is no easy way to implement different color modes for your website if you don't know how to code.
If I want my white themed website to have a dark mode, I currently have to deal with Javascript, cookies or localstorage, custom css.
For a drag en drop page builder, that's not optimal.

### Describe the solution you'd like

Add an option for multi theme colors to the color tab on site settings.
1. Clicking this option will add an extra row of colors.
2. Allows people to select the color pallete they want to edit.
3. A list with created color pallets. In my example, I use numbers, but maybe being able to name each pallete is better.
![example](https://user-images.githubusercontent.com/29916681/214903070-adc1378e-2a20-40e6-a79d-d9059b908acb.png)
4. A widget that you can use to switch between color themes. Preferable is an icon that changes based on the selected theme.
For example, I will have a light mode and dark mode. Light mode is bound to the moon icon and dark mode to the sun icon. Lets say I click it, it will change to a sun icon. The color palette theme will then switch to the colors that belong to the sun icon. 
![Untitled-2](https://user-images.githubusercontent.com/29916681/214904727-0e0c31bd-08d9-46a3-95c4-0a74eb2f0893.png)
![Schermafbeelding 2023-01-26 181938](https://user-images.githubusercontent.com/29916681/214904737-31325b52-8170-4459-8908-5c1dbe09b013.png)

Also, it would be great to have an option to turn on "detect system preference", the first time someone visits the site. They will automatically get the color theme that belongs to their system's color theme. (Dark/White mode for computers and phones.)

Of course, the selected theme has to be saved so that when a user has selected their preferred theme once and come back to the site at a later date, it will still be their preferred theme.

### Describe alternatives you've considered

Coding it myself. Does not work well in Elementor ecosystem, where you just want to drag and drop to create amazing pages.

**Comment by jgarrett414** [comment]
I created a Github account specifically for this reason, to request a similar feature. While there are some plugins that essentially toggle between style sheets the ones i have seen still require a lot of manual CSS styling on a per page basis especially when you have a more complex site. 

I like the idea of having the ability to create multiple global color sets, it would work well for saving color sets to easily select for your needs globally, however i would suggest a different route to styling and fine tuning dark mode or any other "mode". Have it operate in a similar way as breakpoints do within Elementor. And go beyond just color, incorporate fonts as well. 

1. Create as many "stylesheets" for lack of a better term as you like, just like you can create custom breakpoints, 
2. Name them how you want and toggle between them 
3. Fine tune your colors/fonts for every single element and widget for whatever "stylesheet" you have selected.
4. Integrate into the "Dynamic Tags" for a link on a widget, an "action item" for switching between "stylesheets". That way you can make any button, icon list or any element that can be clicked a switch to toggle between "stylesheets". 

This has many use cases beyond just a light/dark mode such as creating multiple style sets for accessibility needs, quickly changing styles for a holiday, or perhaps creating multiple styles for users/visitors to choose their own experience, or assign a style based on user role. 

Elementor should handle all aspects of the design and style of a website, why leave a feature like this up to a separate plugin. And most of the plugins that i have seen that add some sort of Light/dark ability still require a lot of custom CSS to get everything right. They basically just change some of the major things that are mostly at the global level but as for individual widgets and elements those often need to be fine tuned with custom written CSS.
